### ISSUE TITLE
Primitive for normalizing (feature scaling) input data

### ISSUE THREAD

**Issue by itinawi** [HDI-Project/MLPrimitives#82]
I want to create a primitive for normalization of data, or feature scaling so that the input is rescaled to `[-1,1]`

The formula for rescaling is 
`X_rescaled = (X - X.min) / (X.max - X.min)`

The primitive input arguments are:
- `data` (pandas dataframe)
- `column` (string): the column to be rescaled
- `trim_percentage` (float): percentage from the bottom and top to trim
- `inplace=True`: if false, creates a new column called `rescaled_input`

Here are some potential issues with the implementation:

- the implementation will run on historic data. If this primitive were to be used in an online system, we would have to either implement dynamic rescaling of dataset or automatically flag values larger than `min` and `max` as anomalies. Or we can just clip the input and assign large values as `-1` or `1`. The last suggestion makes the most sense, in my opinion.

- distribution of the data. The data may have an outlier (very large value, `e.g. 1234`) and then the remaining values would be somewhere between `-10 and 10`. This would result in bad rescaling. One solution is to trim the lowest and highest 1% values, but what if those were outliers?

**Comment by csala** [comment]
This functionality seems to be mostly covered by [scikit-learn's MinMaxScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html), so I think it would be better to integrate that instead.

The only thing that would really be new would be the trimming of the values, but this could go on a dedicated primitive.

@itinawi if you agree on that, would you mind editing the title and description to have this dedicated only to the trimming, and open a separated issue for the MinMaxScaler integration?

**Comment by csala** [comment]
The MinMaxScaler will be covered in #94 

This can be kept for the trimming.
